src/components/Person.js

```diff
@@ -1,131 +1,127 @@
 
 import {updated_niceties_spinlock} from "./People";
 
-class Person extends React.Component {
-    constructor(props) {
-        super(props)
-        let textValue = '';
-        let anonValue = false;
-        let noReadValue = false;
-        let dataPerson;
-        let foundPerson = false;
-        for (var i = 0; i < props.fromMe.length; i++) {
-            if (props.fromMe[i].target_id === props.data.id) {
-                dataPerson = props.fromMe[i];
-                foundPerson = true;
-                break;
-            }
-        }
-        let dateUpdated;
-        if (foundPerson) {
-            if (dataPerson.date_updated === null) {
-                dateUpdated = '';
-            } else {
-                dateUpdated = dataPerson.date_updated;
-            }
-        }
-        if (foundPerson && store.get("date_updated-" + props.data.id) === null) {
-            if (dataPerson.text !== '' && dataPerson.text !== null) {
-                store.set("nicety-" + props.data.id, dataPerson.text);
-                textValue = dataPerson.text;
-            } else {
-                textValue = '';
-            }
-            store.set("anonymous-" + props.data.id, dataPerson.anonymous);
-            anonValue = dataPerson.anonymous;
-            store.set("no_read-" + props.data.id, dataPerson.no_read);
-            noReadValue = dataPerson.no_read;
-            store.set("date_updated-" + props.data.id, dateUpdated);
-        } else if (foundPerson && store.get("date_updated-" + props.data.id) !== dateUpdated) {
-            if (dataPerson.text !== '' && dataPerson.text !== null) {
-                store.set("nicety-" + props.data.id, dataPerson.text);
-                textValue = dataPerson.text;
-            } else {
-                textValue = '';
-            }
-            store.set("anonymous-" + props.data.id, dataPerson.anonymous);
-            anonValue = dataPerson.anonymous;
-            store.set("no_read-" + props.data.id, dataPerson.no_read);
-            noReadValue = dataPerson.no_read;
-            store.set("date_updated-" + props.data.id, dateUpdated);
-        } else {
-            if (store.get("nicety-" + props.data.id) !== null) {
-                textValue = store.get("nicety-" + props.data.id);
-            }
-            if (store.get("anonymous-" + props.data.id) !== null) {
-                anonValue = store.get("anonymous-" + props.data.id);
-            }
-            if (store.get("no_read-" + props.data.id) !== null) {
-                noReadValue = store.get("no_read-" + props.data.id);
-            }
-        }
+const Person = (props) => {x
+    const [textValue, setTextValue] = useState("");
+    const [anonValue, setAnonValue] = useState(false);
+    const [noReadValue, setNoReadValue] = useState(false);
 
-        this.state = {
-            textValue: textValue,
-            anonValue: anonValue,
-            noReadValue: noReadValue,
-        }
-    }
+    const updateState = () => {
```

Comment:
This is a terrible name for a function but I couldn't describe this function better, as it is long and unwieldy.